samples/cellular/modem_shell/src/link/link.c

```diff
@@ -91,6 +92,24 @@ static struct ncellmeas_data ncellmeas_param_data = {
 	.periodic_interval_given = false,
 };
 
+NRF_MODEM_LIB_ON_CFUN(link_cfun_hook, link_on_modem_cfun, NULL);
+
+static void link_on_modem_cfun(int mode, void *ctx)
+{
+	ARG_UNUSED(ctx);
+
+	switch (mode) {
+	case LTE_LC_FUNC_MODE_NORMAL:
+	case LTE_LC_FUNC_MODE_RX_ONLY:
+	case LTE_LC_FUNC_MODE_ACTIVATE_LTE:
+		(void)lte_lc_modem_events_enable();
```

Comment:
A bit outside of this PR but I've earlier thought that we should probably handle these dynamically made subscriptions in lte_lc when the subscription is done through lte_lc functions.

Reply:
That's a good point. After being enabled once using `lte_lc_modem_events_enable()`, the subscription should stay valid. I can refactor this next week.

Reply:
Refactored.